tsdb/db_test.go

```diff
@@ -6334,3 +6352,54 @@ func compareSeries(t require.TestingT, expected, actual map[string][]tsdbutil.Sa
 		}
 	}
 }
+
+func TestChunkQuerierReadWriteRace(t *testing.T) {
+	db := openTestDB(t, nil, nil)
+	defer func() {
+		require.NoError(t, db.Close())
+	}()
+
+	lbls := labels.FromStrings("foo", "bar")
+
+	ch := make(chan struct{})
+	go func() {
+		defer close(ch)
```

Comment:
I will try to refactor the test.

Reply:
I moved them into `reader` and `writer` functions. Since the number of readers is not fixed and since only `writer` runs in background, I am not using errgroup. Hopefully, the refactored test is simple to reason about.